### ISSUE TITLE
`test_browser` value incorrectly `Safari` instead of `Playwright`

### ISSUE THREAD

**Issue by mgol** [google/karma-webkit-launcher#4]
A follow-up to #2:

I tested the changes from #2 by copying `index.js` from `main` to my local `node_modules/karma-webkit-launcher` and I do see `test_browser` being added, although the value is incorrectly set to `Safari` instead of `Playwright`. Note that I'm using the `playwright-webkit` package, not `playwright` but that shouldn't matter as `ms-playwright` is still part of the path; a relevant part of the `ps axo pid,ppid,tty,user,group,time,pcpu,pmem,args` output:
```
  PID  PPID TTY      USER               GID      TIME  %CPU %MEM ARGS
 9248     1 ??       mgolebiowski        20   0:01.56  38.5  0.4 /Users/mgolebiowski/Library/Caches/ms-playwright/webkit-1751/com.apple.WebKit.WebContent.xpc/Contents/MacOS/com.apple.WebKit.WebContent.Development
 9251     1 ??       mgolebiowski        20   0:00.50  11.6  0.1 /Users/mgolebiowski/Library/Caches/ms-playwright/webkit-1751/com.apple.WebKit.Networking.xpc/Contents/MacOS/com.apple.WebKit.Networking.Development
 9184  9119 ttys024  mgolebiowski        20   0:00.01   0.0  0.0 /bin/bash /Users/mgolebiowski/Library/Caches/ms-playwright/webkit-1751/pw_run.sh http://localhost:9876/?id=86812371 --user-data-dir=/var/folders/ph/_0txk5ls3zxbzkb_7y97_j4r0000gp/T/ec1502b8-4809-461a-8ae6-4db2b99028f6 --headless --disable-gpu
 9189  9184 ttys024  mgolebiowski        20   0:00.75   3.4  0.2 /Users/mgolebiowski/Library/Caches/ms-playwright/webkit-1751/Playwright.app/Contents/MacOS/Playwright http://localhost:9876/?id=86812371 --user-data-dir=/var/folders/ph/_0txk5ls3zxbzkb_7y97_j4r0000gp/T/ec1502b8-4809-461a-8ae6-4db2b99028f6 --headless --disable-gpu
```

**Comment by MarkusBordihn** [comment]
Thanks for the report. Unfortunately I'm unable to reproduce the issue on my Mac or inside the automated tests like https://github.com/google/karma-webkit-launcher/actions/runs/3949422905

I just uploaded a new automated test for "playwright-webkit" but as you already mentioned it should work like the full package.

What is strange in your case, that the process URL not including the `test_browser` parameter and that you get Safari.
The `test_browser` are hard-coded per browser, which means it looks like that Safari was launched instead of Playwright.

The updated version includes an additional check to never use Safari for headless, so let me know if this fixes the issue.
I will publish the updated version to npm in the next hours after all automated tests are passed.

**Comment by MarkusBordihn** [comment]
Even with `playwright-webkit` I was not able to reproduce the issue locally or on an automated test like https://github.com/google/karma-webkit-launcher/actions/runs/3950661176/jobs/6763495364

**Comment by mgol** [comment]
You can try on my PR: https://github.com/jquery/jquery/pull/5190. I added a commit with debug info to a different branch so you can check out this one: https://github.com/mgol/jquery/tree/local/webkit-tests-debug. To test, run:
```
npm install
npx grunt
npx grunt karma:main
```
and watch out for logs like:
```
WebKit 605.1.15 (Mac OS 10.15.7) LOG: 'document.referrer', 'http://localhost:9876/?id=91126886&test_browser=Playwright'
WebKit 605.1.15 (Mac OS 10.15.7) LOG: 'window.location.search', ''
```

As you can see, with the latest `main` I do see `Playwright` as the value but it was `Safari` with an earlier version.

**Comment by MarkusBordihn** [comment]
Thanks for the feedback. I added additional support for automatically using Safari (if Playwright is not available) in the latest version. In the first commit I had a typing error with "&" instead of "&&" which could cause this issue.

I tried to reproduce the issue in your test branch and it looks fine for me.

![image](https://user-images.githubusercontent.com/12183293/213256968-22093e3d-7943-4405-b4af-da6f42abd85a.png)

Let me know if you are able to reproduce the issue with the published version 2.1.0 and I'm happy to take a closer look. Thanks.
